### PR TITLE
Add code needed for evaluating metrics as models

### PR DESCRIPTION
**Pull request by lilacheden**
1. Speed up correlation metrics by not running correlation on each instance
2. Return global scores and disable confidence intervals for evaluate() in eval_utils

### CODE REVIEW

**Comment by assaftibm** [comment]
nice additions

File: src/unitxt/eval_utils.py [hunk 1]
```diff
@@ -1,29 +1,70 @@
...
+    global_scores = {}
     for metric_name in metric_names:
         multi_stream = MultiStream.from_iterables({"test": dataset}, copying=True)
         metrics_operator = SequentialOperator(steps=[metric_name])
+
+        if not compute_conf_intervals:
+            first_step = metrics_operator.steps[0]
+            if isinstance(first_step, MetricPipeline):
+                n_samples_before = first_step.metric.n_resamples
+                first_step.metric.n_resamples = None
```
Comment (matanor) [inline src/unitxt/eval_utils.py:34]: perhaps call this `disable_confidence_interval_calculation` (see [here](https://github.com/IBM/unitxt/blob/main/src/unitxt/metrics.py#L77)) on the metric, its the same, but wraps the inner use of the `n_resamples` field.

File: src/unitxt/eval_utils.py [hunk 2]
```diff
@@ -1,29 +1,70 @@
...
         instances = list(metrics_operator(multi_stream)["test"])
         for entry, instance in zip(dataset, instances):
             entry[metric_name] = instance["score"]["instance"]["score"]
-    return dataset
+
+        if len(instances) > 0:
+            global_scores[metric_name] = instances[0]["score"].get("global", {})
+
+        if not compute_conf_intervals:
+            first_step = metrics_operator.steps[0]
```
Comment (matanor) [inline src/unitxt/eval_utils.py:47]: Please document this hack with a comment, and mention this issue https://github.com/IBM/unitxt/issues/325 in the comment. Also once this PR is merged, please add a reference from the issue to this code. With the motivation being that once the issue 325 is handled, we will remember to go back here and remove this hack.

File: prepare/metrics/kendalltau.py [hunk 3]
```diff
@@ -21,6 +21,31 @@
     metric=KendallTauMetric(),
 )
 
+
+singleton_instance_target = [
```
Comment (matanor) [inline prepare/metrics/kendalltau.py:25]: do you think we need single instance test also for spearmanr?

**Review by matanor — CHANGES_REQUESTED** [review]
hi @lilacheden left some comments in addition to what we talked (avoiding per-instance computation in global metrics) .. pls have a look ..

File: prepare/metrics/kendalltau.py (hunk 3 above)
Comment (lilacheden) [reply prepare/metrics/kendalltau.py]: After the recent change that we discussed, we don't need it for either of them.

File: src/unitxt/metrics.py [hunk 4]
```diff
@@ -989,9 +1006,7 @@ def compute(
         kendall_results = self.kendalltau(references, predictions, variant=self.variant)
         corr = kendall_results.correlation
         return {
-            "score_name": self.main_score,
```
Comment (matanor) [inline src/unitxt/metrics.py:992]: just checking - why are these removed?

**Review by matanor — APPROVED** [review]
LGTM, left one questions, just to make sure something is as expected ..
Thanks!

File: src/unitxt/metrics.py (hunk 4 above)
Comment (lilacheden) [reply src/unitxt/metrics.py:992]: because they are set outside anyway if globalmetric:
```
def _compute(
        self,
        references: List[List[str]],
        predictions: List[str],
        additional_inputs: List[Any],
    ) -> dict:
        result = self.compute(references, predictions, additional_inputs)
        result["score"] = result[self.main_score]
        result["score_name"] = self.main_score
        return result
```
